agent/config/config.go

```diff
@@ -233,20 +268,9 @@ func EnvironmentConfig() Config {
 	disableMetrics := utils.ParseBool(os.Getenv("ECS_DISABLE_METRICS"), false)
 	dockerGraphPath := os.Getenv("ECS_DOCKER_GRAPHPATH")
 
-	reservedMemoryEnv := os.Getenv("ECS_RESERVED_MEMORY")
-	var reservedMemory64 uint64
-	var reservedMemory uint16
-	if reservedMemoryEnv == "" {
-		reservedMemory = 0
-	} else {
-		reservedMemory64, err = strconv.ParseUint(reservedMemoryEnv, 10, 16)
-		if err != nil {
-			log.Warn("Invalid format for \"ECS_RESERVED_MEMORY\" environment variable; expected unsigned integer.", "err", err)
-			reservedMemory = 0
-		} else {
-			reservedMemory = uint16(reservedMemory64)
-		}
-	}
+	reservedMemory := parseEnvVariableInt("ECS_RESERVED_MEMORY", 0)
+
+	cleanupWaitDuration := parseEnvVariableDuration("ECS_ENGINE_CLEANUP_WAIT_DURATION", DefaultConfig().CleanupWaitDuration)
```

Comment:
Can you either remove the default value from `parseEnvVariableDuration` (as [demonstrated](https://github.com/aws/amazon-ecs-agent/pull/286#issuecomment-172711885)) or set this to `0 * time.Millisecond`?  Having it set to a value here will cause problems with `Merge()`.